internal/upserter/upserter.go

```diff
@@ -107,6 +109,37 @@ func doUpsert(ctx context.Context, db *sqlx.DB, logger *zap.Logger, id string, i
 		}
 	}
 
+	upsertSuccess := false
+	maxUpsertRetries := viper.GetInt("crdb.max_retries")
+	dbRetryInterval := viper.GetDuration("crdb.retry_interval")
+
+	for i := 0; i <= maxUpsertRetries; i++ {
+		err = performUpsert(ctx, db, logger, id, upsertRecordFunc, conflictIPs, staleInstanceIPAddresses, newInstanceIPAddresses)
+		if err == nil {
+			if i > 0 {
+				logger.Sugar().Info("DB upsert transaction for instance ", id, "successful on retry attempt #", i)
+			}
+
+			upsertSuccess = true
+
+			break
+		}
+
+		// Exponential backoff would be overkill here, but adding a bit of jitter
+		// to sleep a short time is reasonable
+		jitter := time.Duration(rand.Int63n(int64(dbRetryInterval)))
+		time.Sleep(jitter)
```

Comment:
should this maybe just go into an `else` block?

Reply:
Agree with this recommendation. Updated.